cmd/test/cmd.go

```diff
@@ -156,42 +157,73 @@ func doRun(ctx context.Context, options *Options, ioCtrl *io.Controller, k8sLogg
 
 	builder := buildv2.NewBuilderFromScratch(ioCtrl, nil)
 
-	c := deployCMD.Command{
-		GetManifest: func(path string, fs afero.Fs) (*model.Manifest, error) {
-			return manifest, nil
-		},
-		K8sClientProvider:  k8sClientProvider,
-		Builder:            builder,
-		GetDeployer:        deployCMD.GetDeployer,
-		EndpointGetter:     deployCMD.NewEndpointGetter,
-		DeployWaiter:       deployCMD.NewDeployWaiter(k8sClientProvider, k8sLogger),
-		CfgMapHandler:      configmapHandler,
-		Fs:                 fs,
-		PipelineCMD:        pc,
-		AnalyticsTracker:   at,
-		IoCtrl:             ioCtrl,
-		K8sLogger:          k8sLogger,
-		IsRemote:           env.LoadBoolean(constants.OktetoDeployRemote),
-		RunningInInstaller: config.RunningInInstaller(),
-	}
-	if err = c.Run(ctx, &deployCMD.Options{
-		Manifest:         manifest,
-		ManifestPathFlag: options.ManifestPathFlag,
-		ManifestPath:     options.ManifestPath,
-		Name:             options.Name,
-		Namespace:        options.Namespace,
-		K8sContext:       options.K8sContext,
-		Variables:        options.Variables,
-		Build:            false,
-		Dependencies:     false,
-		Timeout:          options.Timeout,
-		RunWithoutBash:   false,
-		RunInRemote:      manifest.Deploy.Remote,
-		Wait:             true,
-		ShowCTA:          false,
-	}); err != nil {
-		oktetoLog.Error("deploy failed: %s", err.Error())
-		return err
+	kubeClient, _, err := k8sClientProvider.Provide(okteto.GetContext().Cfg)
+	if err != nil {
+		return fmt.Errorf("could not instantiate kuberentes client: %w", err)
+	}
+
+	namer := deployCMD.Namer{
+		KubeClient:   kubeClient,
+		Workdir:      initialCWD,
```

Comment:
Not sure we should use `initialCWD`. That is calculated before the working directory might change in line `137` (internally, `UpdateCWDtoManifestPath` changes the current working directory). In the case of deploy, we get the current working directory after changing it